crates/storage/provider/src/transaction.rs

```diff
@@ -480,7 +480,8 @@ where
 
         // Insert the blocks
         for block in blocks {
-            self.insert_block(block)?;
+            let (block, senders) = block.into_components();
+            insert_canonical_block(self.deref_mut(), block, Some(senders)).unwrap();
```

Comment:
this can fail due to a DB error, 
can we propagate the error instead of panicking?
But we also had this previously

Reply:
Yes and Yes and Yes.

`unwrap` removed and we now propagate error.